react/react-router/lesson.md

```diff
@@ -1 +1,147 @@
-## React router
\ No newline at end of file
+Webové aplikace o více stránkách jsou v dnešní době standard. Představte si muzeum s jedním jediným obrazem - pořád to může být dechberoucí zážitek, ale omrzí výrazně rychleji. Jednostránkové weby rozhodně mají své místo, dnes se však naučíme způsob, jak naši aplikaci rozšířit o další stránky a posunout ji na vyšší úroveň.
+
+## Single-page application (SPA)
+
+Je typ aplikace, u které se obsah stránky mění dynamicky za jejího běhu, aniž by bylo třeba přenačítat celou stránku znovu. Výhodou je rychlejší a hladší navigace stránkou - uživatel nemusí čekat na načítaní každé jednotlivé podstránky. Další plus je snížená zátěž na server, protože místo odpovědi pro každou navštívenou stránku stačí uživateli odpovědět jen jednou a o navigaci se postará jeho prohlížeč. SPA mají i své nevýhody - například horší možnosti SEO optimalizace, ale to už je nad rámec této lekce.
+
+## React Router
+
+Neexistuje žádný magický přepínač, kterým bychom rozhodli, jestli z aplikace uděláme SPA nebo ne. Takovou funkcionalitu někdo musel nejdříve vymyslet, implementovat a následně aktivně používat v kódu. Naštěstí žijeme v krásné době, kde je taková implementace hotová a připravená k použití. K práci s navigací v našich React aplikacích budeme používat knihovnu [React Router](https://reactrouter.com/web/guides/quick-start). Nainstalujeme si ji jako závislost přes npm a následně z ní můžeme používat připravené komponenty přímo v našem kódu.
+
+### Instalace a používání knihovny
+
+Do existujícího projektu můžeme nainstalovat React Router knihovnu přest terminál pomocí npm.
+
+```sh
+npm install react-router-dom
+```
+
+Nyní máme v projektu k dispozici celou škálu komponent, se kterými můzeme pracovat. Stačí si je správně naimportovat a použít jako kteroukoliv jinou komponentu.
+
+```js
+import React from 'react';
+import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
+
+export default function App() {
+  return (
+    <Router>
+      <div>
+        <nav>
+          <ul>
+            <li>
+              <Link to="/">Home</Link>
+            </li>
+            <li>
+              <Link to="/about">About</Link>
+            </li>
+            <li>
+              <Link to="/users">Users</Link>
+            </li>
+          </ul>
+        </nav>
+
+        {/* A <Switch> looks through its children <Route>s and
+            renders the first one that matches the current URL. */}
+        <Switch>
+          <Route path="/about">
+            <About />
+          </Route>
+          <Route path="/users">
+            <Users />
+          </Route>
+          <Route path="/">
+            <Home />
+          </Route>
+        </Switch>
+      </div>
+    </Router>
+  );
+}
+
+function Home() {
```

Comment:
Napadá mě, nebylo by lepší upřednostnit konzistenci s ostatními lekcemi a komponenty psát spíš přes arrow funkce nehledě na to, že v dokumentaci routeru je to jinak?

Reply:
Nad tim jsem taky premyslel. Je tam prave to rozpolceni mezi 1) konzistence s ostatnimi lekcemi .. versus 2) konzistence s dokumentaci, kterou si k tomu hned otevrou a budou potencialne zmatene, ze je to tam jinak a v jinem poradi? (consty bych tusim musel dat nahoru).
Myslis, ze konzistence s ostatnimi lekcemi lepsi?

Reply:
Nejsem si jistý. Nechávám to na tobě.

Vidím to tak, že dokumentace vysvětluje, jak to funguje a použití `function` je implementační detail.

Lekce za mě říká, jak to mají účastnice psát ve svých projektech. Chceš je takhle v závěru přeučovat z arrow na klasické funkce?

Reply:
Preucovat rozhodne ne no. :-D

Myslim, ze mas pravdu. V prikladu tam nekde uvadim `const Home`, ale kdyz uz to nektere budou kopirovat z lekce, tak at si tam nezanaseji zbytecne `function`.

Upravim to.